backend/src/controllers/tasks.controller.ts

```diff
@@ -74,22 +85,54 @@ async function updateTask(req: Request, res: Response) {
 async function deleteTask(req: Request, res: Response) {
   try {
     const { idTask } = req.params;
+    const { idProject } = req.params;
+
+    const projectCheck = await Project.findById(idProject);
 
-    const taskDetail = await Task.findByIdAndDelete(idTask);
-    if (!taskDetail) throw new Error('Not found a task');
+    if (!projectCheck) throw new Error('Not found a Project');
 
-    res.status(200).send(taskDetail);
+    const projectDetail = await Project.findByIdAndUpdate(
+      idProject,
+      { $pull: { tasks: { _id: idTask } } },
+      { new: true }
+    );
+
+    if (!projectDetail) throw new Error('Not found a Project');
+
+    if (projectCheck.tasks.length === projectDetail.tasks.length) throw new Error('Not found a Task');
+
+    res.status(200).send(projectDetail.tasks);
   } catch (error) {
     res.status(400).send((error as Error).message);
   }
 }
 
 async function deleteAllTask(req: Request, res: Response) {
   try {
-    const taskDetail = await Task.deleteMany();
-    if (!taskDetail.deletedCount) throw new Error('Not found a task');
+    const { idProject } = req.params;
+    const projectDetail = await Project.findByIdAndUpdate(idProject, { $pull: { tasks: {} } });
+
+    if (!projectDetail) throw new Error('Not found a Project');
+    if (!projectDetail.tasks.length) throw new Error('Not found a Task');
+
+    res.status(200).send(`Deleted completed successfully`);
+  } catch (error) {
+    res.status(400).send((error as Error).message);
+  }
+}
+
+async function getAllComments(req: Request, res: Response) {
+  try {
+    const { idTask } = req.params;
+    const { idProject } = req.params;
 
-    res.status(200).send(`Deleted completed successfully, Deleted ${taskDetail.deletedCount} Tasks`);
+    const projectDetail = await Project.findById(idProject, { tasks: { $elemMatch: { _id: idTask } } });
+
+    if (!projectDetail) throw new Error('Not found a Project');
+    if (!projectDetail.tasks.length) throw new Error('Not found a Task');
+    if (!projectDetail.tasks[0].comments.length) throw new Error('Not found a Comment');
+
+    res.status(200).send(projectDetail.tasks[0].comments);
```

Comment:
please do it like this 
const comments = projectDetail.tasks[0].comments
res.status(200).send(comments)